### PR TITLE
Pause/Resume Settings and Streams in Acq Plugins

### PR DESCRIPTION
**Pull request by amuskens**
…t settings from changing during acquisition.

### CODE REVIEW

File: plugins/timelapse.py [hunk 1]
```diff
@@ -267,8 +267,7 @@ def start(self):
 
     def acquire(self, dlg):
         main_data = self.main_app.main_data
-        str_ctrl = main_data.tab.value.streambar_controller
```
Comment (pieleric) [inline plugins/timelapse.py]: You cannot remove these lines, as they don't do the same thing as `.pause()`. These lines ensure that no stream is live.

File: src/odemis/gui/plugin/__init__.py [hunk 2]
```diff
@@ -559,6 +559,22 @@ def on_close(self, btnid, _):
         self.streambar_controller.clear()
         self.EndModal(btnid)
 
+    def _pause_settings(self):
+        """ Pause the settings of the GUI and save the values for restoring them later """
```
Comment (pieleric) [inline src/odemis/gui/plugin/__init__.py]: If it's supposed to be called by external code, don't make the method "private". => `pauseSettings`/`resumeSettings`.

Comment (pieleric) [reply src/odemis/gui/plugin/__init__.py]: Also, the docstring can be improved. In particular, it does _not_ save the values (ie, if they are changed, on resume the values will be displayed changed). I'd suggest something like:
`Pause the settings widgets. They will be disabled and the value frozen even when the VAs are changed`.

**Comment by amuskens** [comment]
Added the changes.

File: doc/develop/plugins.rst [hunk 3]
```diff
@@ -278,6 +278,7 @@ When that entry is selected, it shows an acquisition window and then acquire
             ccd = self.main_data.ccd
             exp = self.exposureTime.value
             ccd.exposureTime.value = exp
+            dlg.pauseSettings()	# Freezes the setting and stream controls in window
```
Comment (pieleric) [inline doc/develop/plugins.rst]: You need to also _document_ these new methods (after documentation `setAcquisitionInfo()`).

File: plugins/tileacq.py [hunk 4]
```diff
@@ -655,6 +655,7 @@ def acquire(self, dlg):
         main_data = self.main_app.main_data
         str_ctrl = main_data.tab.value.streambar_controller
         stream_paused = str_ctrl.pauseStreams()
+        dlg.pauseSettings()
```
Comment (pieleric) [inline plugins/tileacq.py]: Also need to call `resumeSettings()`. Probably line 764 (just after `dlg.Destroy()`) and in the `CancelledError` handler. And you need to change the `return`s from the cancellation place to `raise CancelledError`.

File: plugins/tileacq.py [hunk 5]
```diff
@@ -761,6 +763,7 @@ def acquire(self, dlg):
             # End of the (completed) acquisition
             if ft._task_state != CANCELLED:
                 dlg.Destroy()
+                dlg.resumeSettings()
```
Comment (pieleric) [inline plugins/tileacq.py]: It should be the opposite, only call `resumeSettings()` if it's cancelled. => in a else block.

File: plugins/tileacq.py [hunk 6]
```diff
@@ -478,7 +478,9 @@ def _cancel_acquisition(self, future):
             future.running_subf.cancel()
             logging.debug("Acquisition cancelled.")
 
-        return True
+        dlg.resumeSettings()
```
Comment (pieleric) [inline plugins/tileacq.py]: Please don't put this in the canceller. It was fine like this. You need to change the acquisition code, in all the blocks that check for cancel to look like:
```
 if ft._task_state == CANCELLED:
      raise CancelledError("Acquisition cancelled")
```
And at the end, the exception handler would look like:
```         
except CancelledError:
    logging.debug("Acquisition cancelled")
    dlg.resumeSettings()
```

File: doc/develop/plugins.rst [hunk 7]
```diff
@@ -291,8 +301,9 @@ When that entry is selected, it shows an acquisition window and then acquire
                 f.set_progress(end=time.time() + left)
                 d.append(ccd.data.get())
                 if f.cancelled():
-                	dlg.resumeSettings() # Unfreezes the setting and stream controls in window
-                    return
+                    # Unfreezes the setting and stream controls in window
+	            	dlg.resumeSettings()
```
Comment (pieleric) [inline doc/develop/plugins.rst]: Please, don't insert tabs, only spaces. (Also in the rst files).

**Comment by amuskens** [comment]
Changes made.

**Pull request merged**